datafusion/sql/tests/cases/diagnostic.rs

```diff
@@ -274,3 +279,36 @@ fn test_ambiguous_column_suggestion() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_sum_coerce_type() -> Result<()> {
+    let mut sum = Sum::new();
+    let span = Span::new(Location::new(1, 1), Location::new(2, 1));
+    let span_for_test = span;
+    sum.spans_mut().add_span(span);
+    let datatype_vec = vec![DataType::Date64];
+    let res = sum.coerce_types(&datatype_vec);
+    let err = res.expect_err("Expected an error, but got success");
```

Comment:
Can you use the same setup as other tests in this file? e.g. `get_spans`, `run_query`, etc.